### PR TITLE
Fix #1983 app does not crash on selecting or bookmarking a track having empty fields

### LINKED ISSUE #1983: App crashes on selecting or bookmarking an event having a null value
**Actual Behaviour**

App crashes on bookmarking or selecting an event in sessionDetails Activity having a null value

**Expected Behaviour**

App should not crash

**Steps to reproduce it**

Select schedule. Click on open tech. click on the  first event.

**LogCat for the issue**

java.lang.NullPointerException: Attempt to invoke virtual method 'java.lang.String org.fossasia.openevent.data.Track.getFontColor()' on a null object reference

**Screenshots of the issue**

ezgif com-video-to-gif 19

**Would you like to work on the issue?**

Yes

### PR DESCRIPTION
**Pull request by nikit19**
Fixes #1983 

Changes: app does not crash on selecting or bookmarking a track having empty fields

Screenshots for the change: 
ezgif com-video-to-gif 20

### CODE REVIEW

**Review by scandindian — APPROVED** [review]
This works fine but the title of pull request is misleading.

**Comment by nikit19** [comment]
is it better now?

File: android/app/src/main/java/org/fossasia/openevent/activities/SessionDetailActivity.java [hunk 1]
```diff
@@ -464,9 +464,11 @@ protected void onResume() {
 
     private void loadSession(Session session) {
         this.session = session;
-        this.trackColor = Color.parseColor(session.getTrack().getColor());
-        this.fontColor = Color.parseColor(session.getTrack().getFontColor());
-        this.darkColor = Views.getDarkColor(Color.parseColor(session.getTrack().getColor()));
+        if(session.getTrack()!=null) {
```
Comment (iamareebjamal) [inline android/app/src/main/java/org/fossasia/openevent/activities/SessionDetailActivity.java]: Please add a space after `if`

File: android/app/src/main/java/org/fossasia/openevent/adapters/SessionsListAdapter.java [hunk 2]
```diff
@@ -236,7 +236,7 @@ public void onBindViewHolder(final SessionViewHolder holder, final int position)
                 } else {
                     Snackbar.make(holder.sessionCard, R.string.removed_bookmark, Snackbar.LENGTH_SHORT).show();
                 }
-            } else {
+            } else if(track!=null){
```
Comment (iamareebjamal) [inline android/app/src/main/java/org/fossasia/openevent/adapters/SessionsListAdapter.java]: Please add space around `!=` and after `)`

File: android/app/src/main/java/org/fossasia/openevent/adapters/SessionsListAdapter.java [hunk 3]
```diff
@@ -220,7 +220,7 @@ public void onBindViewHolder(final SessionViewHolder holder, final int position)
         final int sessionId = session.getId();
 
         holder.sessionBookmarkIcon.setOnClickListener(v -> {
-            if (session.getIsBookmarked()) {
+            if (session.getIsBookmarked() && track!=null) {
```
Comment (iamareebjamal) [inline android/app/src/main/java/org/fossasia/openevent/adapters/SessionsListAdapter.java]: Here as well

**Review by iamareebjamal — CHANGES_REQUESTED** [review]

**Comment by nikit19** [comment]
@iamareebjamal done with the requested changes

File: android/app/src/main/java/org/fossasia/openevent/adapters/SessionsListAdapter.java [hunk 4]
```diff
@@ -236,7 +236,7 @@ public void onBindViewHolder(final SessionViewHolder holder, final int position)
                 } else {
                     Snackbar.make(holder.sessionCard, R.string.removed_bookmark, Snackbar.LENGTH_SHORT).show();
                 }
-            } else {
+            } else if (track != null) {
```
Comment (iamareebjamal) [inline android/app/src/main/java/org/fossasia/openevent/adapters/SessionsListAdapter.java]: This logic can be reduced by using `if (track == null) return;` at the start of on click listener

**Review by iamareebjamal — CHANGES_REQUESTED** [review]

File: android/app/src/main/java/org/fossasia/openevent/adapters/SessionsListAdapter.java (hunk 4 above)
Comment (nikit19) [reply android/app/src/main/java/org/fossasia/openevent/adapters/SessionsListAdapter.java]: done with the requested changes

**Review by iamareebjamal — DISMISSED** [review]

**Comment by iamareebjamal** [comment]
@nikit19 Please resolve conflicts

**Comment by nikit19** [comment]
@iamareebjamal conflicts are resolved now

**Review by iamareebjamal — APPROVED** [review]

**Pull request merged**